src/libdredd/src/mutate_visitor.cc

```diff
@@ -258,6 +258,30 @@ bool MutateVisitor::HandleUnaryOperator(clang::UnaryOperator* unary_operator) {
     return true;
   }
 
+  // There is no useful way to mutate this expression.
+  if (optimise_mutations_ && ((unary_operator->getOpcode() == clang::UO_Minus &&
+                               (MutationReplaceExpr::ExprIsEquivalentToInt(
+                                    *unary_operator->getSubExpr(), 1,
+                                    compiler_instance_.getASTContext()) ||
+                                MutationReplaceExpr::ExprIsEquivalentToFloat(
+                                    *unary_operator->getSubExpr(), 1,
```

Comment:
For the float cases, please use a double literal - so 1.0 in this case. Please do the same for other floating-point checks.

Reply:
Done.